### PR TITLE
Task/platform 104 common application

### PR DESCRIPTION
**Pull request by solomontang**
- Add multipage common application schemas
- submit/save access request onPageChange
- add print button to top of EditAccessRequestContainer

### CODE REVIEW

File: src/containers/access/src/sagas/submitAccessRequest.js [hunk 1]
```diff
@@ -86,9 +93,18 @@ function* submitAccessRequestWorker(action :SequenceAction) :Saga<WorkerResponse
...
+    const localEntity = Map({
+      [TYPE]: [type || 'Common Application'],
+      [FORM_DATA]: [formDataStr],
+      [RJSF_JSON_SCHEMA]: [schemaStr],
+      [RJSF_UI_SCHEMA]: [uiSchemaStr],
+      [REQUEST_DATE_TIME]: [now],
+      [OPENLATTICE_ID_FQN]: submitAccessResponse.data
+    });
+    response = { data: localEntity };
+    debugger;
```
Comment (bergieboy) [inline src/containers/access/src/sagas/submitAccessRequest.js]: should this be here?

File: src/forms/common/CommonApplicationForm.js [hunk 2]
```diff
@@ -1,45 +1,119 @@
...
-const { isPending, isSuccess } = ReduxUtils;
+const { isPending } = ReduxUtils;
+const { getEntityKeyId } = DataUtils;
+const { isValidUUID } = ValidationUtils;
+
+const ActionRow = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 0 30px 30px 30px;
```
Comment (bergieboy) [inline src/forms/common/CommonApplicationForm.js:31]: sort

File: src/forms/common/CommonApplicationForm.js [hunk 3]
```diff
@@ -1,45 +1,119 @@
...
+          return (
+            <>
+              <Form
+                  ref={formRef}
+                  hideSubmit
+                  isSubmitting={pending}
+                  formData={pagedData}
+                  onSubmit={onNext}
+                  schema={schemas[page]}
+                  uiSchema={uiSchemas[page]} />
```
Comment (bergieboy) [inline src/forms/common/CommonApplicationForm.js:99]: sort

**Review by giuliacampana — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -143,5 +143,8 @@
     "test": "jest --config config/jest/jest.config.js --coverage=false",
     "test:travis": "jest --maxWorkers=4 --config config/jest/jest.config.js",
     "test:watch": "jest --config config/jest/jest.config.js --watch --coverage=false"
+  },
+  "dependencies": {
+    "react-to-print": "^2.12.3"
```
Comment (UnsungHero97) [inline package.json:148]: You need to add this to `externals` in the webpack config.

```
externals: [
  externals({
    allowlist: ['react-to-print'],
  }),
]
```

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: src/containers/access/src/EditAccessRequestContainer.js [hunk 5]
```diff
@@ -1,6 +1,12 @@
 // @flow
+import { useRef } from 'react';
 
+import styled from 'styled-components';
+import { faPrint } from '@fortawesome/pro-regular-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { IconButton } from 'lattice-ui-kit';
 import { DataUtils, ReduxUtils } from 'lattice-utils';
+import { useReactToPrint } from 'react-to-print';
```
Comment (UnsungHero97) [inline src/containers/access/src/EditAccessRequestContainer.js:9]: No change required, just a thought. Maybe the parent app should be responsible for this dependency, and all the module does is pass along the ref in a `onPrint` callback or something.

File: src/containers/access/src/reducers/submitAccessRequestReducer.js [hunk 6]
```diff
@@ -19,7 +19,7 @@ export default function submitAccessRequestReducer(state :Map<*, *>, action :Seq
   return submitAccessRequest.reducer(state, action, {
     REQUEST: () => state.setIn([SUBMIT_ACCESS_REQUEST, REQUEST_STATE], RequestStates.PENDING),
     SUCCESS: () => state
-      .merge(action.value)
+      .set('accessRequest', action.value)
```
Comment (UnsungHero97) [inline src/containers/access/src/reducers/submitAccessRequestReducer.js:22]: You created the constant, might as well use it.

File: src/containers/access/src/sagas/submitAccessRequest.js [hunk 7]
```diff
@@ -86,9 +93,23 @@ function* submitAccessRequestWorker(action :SequenceAction) :Saga<WorkerResponse
     );
 
     if (submitAccessResponse.error) throw submitAccessResponse.error;
-    response = { data: submitAccessResponse.data };
-
-    yield put(submitAccessRequest.success(action.id));
+    const localEntity = Map({
+      // $FlowFixMe invalid-computed-prop
+      [TYPE]: [type || 'Common Application'],
```
Comment (UnsungHero97) [inline src/containers/access/src/sagas/submitAccessRequest.js:98]: No change required. `|| 'Common Application'` doesn't make sense as a default in the future.

File: src/forms/common/schemas/page1Schemas.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+      },
+      address: {
+        classNames: 'column-span-12',
+      },
+      phoneNumber: {
+        classNames: 'column-span-6',
+      },
+      email: {
+        classNames: 'column-span-6',
+      },
```
Comment (UnsungHero97) [inline src/forms/common/schemas/page1Schemas.js:86]: No change required, just a thought. You can pull this out and reuse it for all 3.

File: src/forms/common/schemas/page1schemas.js [hunk 9]
```diff
@@ -0,0 +1,131 @@
+export const page1Schema = {
```
Comment (UnsungHero97) [inline src/forms/common/schemas/page1schemas.js:1]: This is a duplicate.

**Review by UnsungHero97 — APPROVED** [review]

**Review by giuliacampana — APPROVED** [review]

**Pull request merged**